File: Game/src/main/java/net/winternetwork/bedwars/game/module/map/MapModule.java
package net.winternetwork.bedwars.game.module.map;

import lombok.Getter;
import net.winternetwork.bedwars.api.module.Module;
import net.winternetwork.bedwars.api.module.ModulePriority;
import net.winternetwork.bedwars.game.module.map.command.MapSetupCommand;

public class MapModule extends Module {

    @Getter
    private MapManager mapManager;

    public MapModule() {
        super("Mapa", ModulePriority.NORMAL);
    }

    @Override
    public void init() {
        mapManager = new MapManager();

        getLogger().info("Carregando mapas...");
        mapManager.loadAll(Core.getInstance().getMapsConfig());

        getLogger().info("Registrando comandos...");
        Core.getInstance()
                .registerCommands(new MapSetupCommand(mapManager));
    }
}

File: Game/src/main/kotlin/net/winternetwork/bedwars/modules/maps/MapModule.kt
package net.winternetwork.bedwars.modules.maps

import net.winternetwork.bedwars.api.config.YamlConfig
import net.winternetwork.bedwars.api.module.Module
import net.winternetwork.bedwars.game.game
import net.winternetwork.bedwars.modules.maps.command.MapCommands
import net.winternetwork.bedwars.modules.maps.manager.MapManager

class MapModule : Module("Maps") {

    private val config: YamlConfig by lazy {
        YamlConfig(
            game,
            dataFolder,
            "maps.yml"
        )
    }

    override val manager = MapManager(config)

    override fun init() {
        log("Registrando comandos...")
        game.commands(MapCommands(manager))
    }
}